packages/tasit-contracts/src/scripts/populateDecentralandContracts.js

```diff
@@ -78,263 +213,184 @@ const createEstate = async (estateContract, landContract, estate, wallet) => {
       const { args } = data;
       resolve(args._estateId);
     });
+
+    setTimeout(() => {
+      estateContract.unsubscribe();
+      reject();
+    }, EVENTS_TIMEOUT);
   });
 
   await action.waitForNonceToUpdate();
-  console.log("created id =", estateId);
+  console.log(`created id ${estateId}`);
 
   return estateId;
 };
 
-const createEstates = async (estateContract, landContract, estates, wallet) => {
+const createEstates = async estates => {
+  console.log("Creating estates...");
+
   const estateIds = [];
   for (let estate of estates) {
-    const id = await createEstate(estateContract, landContract, estate, wallet);
-    estateIds.push(id);
+    try {
+      const id = await createEstate(estate);
+      estateIds.push(id);
+    } catch (error) {
+      console.log("Estate creation failed");
+    }
   }
   return estateIds;
 };
 
-const approveMarketplace = async (
-  marketplaceAddress,
-  landContract,
-  estateContract,
-  assetsOwnerWallet
-) => {
+const approveMarketplace = async () => {
+  console.log("Approving Marketplace...");
+
   // Set false to remove approval
   const authorized = true;
 
-  estateContract.setWallet(assetsOwnerWallet);
+  estateContract.setWallet(sellerWallet);
   const estateApproval = estateContract.setApprovalForAll(
-    marketplaceAddress,
+    MARKETPLACE_ADDRESS,
     authorized,
     gasParams
   );
   await estateApproval.waitForNonceToUpdate();
 
-  landContract.setWallet(assetsOwnerWallet);
+  landContract.setWallet(sellerWallet);
   const landApproval = landContract.setApprovalForAll(
-    marketplaceAddress,
+    MARKETPLACE_ADDRESS,
     authorized,
     gasParams
   );
   await landApproval.waitForNonceToUpdate();
 };
 
-const placeEstatesSellOrders = async (
-  marketplace,
-  estateAddress,
-  estateIds,
-  priceInWei,
-  expireAt,
-  sellerWallet
-) => {
-  marketplace.setWallet(sellerWallet);
-  for (let assetId of estateIds) {
-    const action = marketplace.createOrder(
-      estateAddress,
-      assetId,
-      priceInWei,
-      expireAt,
-      gasParams
-    );
-    await action.waitForNonceToUpdate();
+function getRandomInt(min, max) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
```

Comment:
Why is this step necessary?

Reply:
`getRandonInt`? I'm using that for assets prices.

Reply:
No, the `Math.floor`

Reply:
Oh, I've just copied this function from here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#Getting_a_random_integer_between_two_values
Seems that is a way to parse the number to integer.

Reply:
Ah ok - makes sense. Resolving this!